### PR TITLE
Update AKS module with default configuration

### PR DESCRIPTION
**Pull request by sumanthreddy29**
This pull request includes multiple changes to the Terraform configuration for Azure, primarily focusing on enhancing the Kubernetes cluster setup and adding new variables for better configurability. The most important changes include updates to the `README.md` for new environment variables, modifications to the main Terraform files to support auto-scaling and network policies, and updates to the input schema and variable definitions.

### Documentation Updates:
* [`modules/terraform/azure/README.md`](diffhunk://#diff-6408a79a9cdb905ab9327763a955f3f5cd4d0af74ddb77f688f08a7f82e27ed0L19-R25): Added new environment variables `NETWORK_POLICY` and `EBPF_DATA_PLANE` for performance evaluation scenarios. (`[[1]](diffhunk://#diff-6408a79a9cdb905ab9327763a955f3f5cd4d0af74ddb77f688f08a7f82e27ed0L19-R25)`, `[[2]](diffhunk://#diff-6408a79a9cdb905ab9327763a955f3f5cd4d0af74ddb77f688f08a7f82e27ed0R64-R73)`)

### Terraform Configuration Enhancements:
* [`modules/terraform/azure/aks/main.tf`](diffhunk://#diff-4089bc0576048d82e335da51ad0374e2e9c0ae37db3d65c3c784e97871651fd7R31): Enabled auto-scaling and added an auto-scaler profile and automatic channel upgrade settings for the Kubernetes cluster. (`[[1]](diffhunk://#diff-4089bc0576048d82e335da51ad0374e2e9c0ae37db3d65c3c784e97871651fd7R31)`, `[[2]](diffhunk://#diff-4089bc0576048d82e335da51ad0374e2e9c0ae37db3d65c3c784e97871651fd7R42-R47)`)
* [`modules/terraform/azure/aks/main.tf`](diffhunk://#diff-4089bc0576048d82e335da51ad0374e2e9c0ae37db3d65c3c784e97871651fd7L52-L53): Removed deprecated settings `oidc_issuer_enabled` and `workload_identity_enabled`. (`[modules/terraform/azure/aks/main.tfL52-L53](diffhunk://#diff-4089bc0576048d82e335da51ad0374e2e9c0ae37db3d65c3c784e97871651fd7L52-L53)`)

### Variable and Schema Updates:
* [`modules/terraform/azure/aks/variables.tf`](diffhunk://#diff-e0b25de6470cd5774403141e8997adee371642381fd8430d6dd467f86d13b59eL44-R45): Updated default values for `network_plugin` and `network_plugin_mode` and added new optional variables `aks_network_policy` and `aks_ebpf_data_plane`. (`[modules/terraform/azure/aks/variables.tfL44-R45](diffhunk://#diff-e0b25de6470cd5774403141e8997adee371642381fd8430d6dd467f86d13b59eL44-R45)`)
* [`modules/terraform/azure/azure_input_schema.json`](diffhunk://#diff-4f17a00034a463e9bcadcda2dcc8f0e03cd95603ec9329474618c9f7aad5deaeR13-R18): Added new fields `aks_network_policy` and `aks_ebpf_data_plane` to the input schema. (`[modules/terraform/azure/azure_input_schema.jsonR13-R18](diffhunk://#diff-4f17a00034a463e9bcadcda2dcc8f0e03cd95603ec9329474618c9f7aad5deaeR13-R18)`)
* [`modules/terraform/azure/main.tf`](diffhunk://#diff-0f7f598ee8901914ae02cf0a5c9c9c1237e1bbe208688856f72a3af4e1cc12bbR5-R6): Updated local variables to include new network policy settings and modified the logic for merging configurations. (`[[1]](diffhunk://#diff-0f7f598ee8901914ae02cf0a5c9c9c1237e1bbe208688856f72a3af4e1cc12bbR5-R6)`, `[[2]](diffhunk://#diff-0f7f598ee8901914ae02cf0a5c9c9c1237e1bbe208688856f72a3af4e1cc12bbL17-R46)`)
* [`modules/terraform/azure/variables.tf`](diffhunk://#diff-a70d527b3b015f3aa3103f024416bbf9c1e2828c82890a61a97316c23e5402efR7-R8): Added new optional variables `aks_network_policy` and `aks_ebpf_data_plane` to the JSON input variable definition. (`[modules/terraform/azure/variables.tfR7-R8](diffhunk://#diff-a70d527b3b015f3aa3103f024416bbf9c1e2828c82890a61a97316c23e5402efR7-R8)`)

### CODE REVIEW

**Review by anson627 — APPROVED** [review]

**Review by anson627 — APPROVED** [review]

File: jobs/competitive-test.yml [hunk 1]
```diff
@@ -86,21 +86,21 @@ jobs:
...
-      engine_input: ${{ parameters.engine_input }}
-  - template: /steps/publish-results.yml
-    parameters:
-      cloud: ${{ parameters.cloud }}
-      topology: ${{ parameters.topology }}
-      engine: ${{ parameters.engine }}
-      regions: ${{ parameters.regions }}
-      engine_input: ${{ parameters.engine_input }}
-      credential_type: ${{ parameters.credential_type }}
+  # - template: /steps/execute-tests.yml
```
Comment (anson627) [inline jobs/competitive-test.yml:89]: uncomment

File: pipelines/perf-eval/apiserver-benchmark-virtualnodes10-pods100.yml [hunk 2]
```diff
@@ -54,9 +54,11 @@ stages:
             workload-low:
               flowcontrol: "workload-low:1000"
               extra_benchmark_subcmd_args: ""
+              sku_tier: Free
```
Comment (sumanthreddy29) [inline pipelines/perf-eval/apiserver-benchmark-virtualnodes10-pods100.yml:57]: revert this change

**Review by alyssa1303 — APPROVED** [review]
Please run an internal test before merging
